diff --git a/include/ftxui/component/screen_interactive.hpp b/include/ftxui/component/screen_interactive.hpp
--- a/include/ftxui/component/screen_interactive.hpp
+++ b/include/ftxui/component/screen_interactive.hpp
@@ -37,6 +37,7 @@ class ScreenInteractive : public Screen {
 
   // Options. Must be called before Loop().
   void TrackMouse(bool enable = true);
+  void ForceHandleCtrlC(bool enable = true);
 
   // Return the currently active screen, nullptr if none.
   static ScreenInteractive* Active();
@@ -82,6 +83,7 @@ class ScreenInteractive : public Screen {
                     bool use_alternative_screen);
 
   bool track_mouse_ = true;
+  bool force_handle_ctrl_c_ = true;
 
   Sender<Task> task_sender_;
   Receiver<Task> task_receiver_;
diff --git a/src/ftxui/component/screen_interactive.cpp b/src/ftxui/component/screen_interactive.cpp
--- a/src/ftxui/component/screen_interactive.cpp
+++ b/src/ftxui/component/screen_interactive.cpp
@@ -399,6 +399,10 @@ void ScreenInteractive::TrackMouse(bool enable) {
   track_mouse_ = enable;
 }
 
+void ScreenInteractive::ForceHandleCtrlC(bool enable) {
+  force_handle_ctrl_c_ = enable;
+}
+
 /// @brief Add a task to the main loop. 
 /// It will be executed later, after every other scheduled tasks.
 /// @ingroup component
@@ -543,6 +547,9 @@ void ScreenInteractive::Install() {
   // Install signal handlers to restore the terminal state on exit. The default
   // signal handlers are restored on exit.
   for (const int signal : {SIGTERM, SIGSEGV, SIGINT, SIGILL, SIGABRT, SIGFPE}) {
+    if (signal == SIGINT && !force_handle_ctrl_c_) {
+      continue;
+    }
     InstallSignalHandler(signal);
   }
 
diff --git a/src/ftxui/component/terminal_input_parser.cpp b/src/ftxui/component/terminal_input_parser.cpp
--- a/src/ftxui/component/terminal_input_parser.cpp
+++ b/src/ftxui/component/terminal_input_parser.cpp
@@ -176,7 +176,6 @@ TerminalInputParser::Output TerminalInputParser::Parse() {
   }
 
   switch (Current()) {
-    case 24:  // CAN NOLINT
     case 26:  // SUB NOLINT
       return DROP;
 
