diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,6 +63,7 @@ add_library(dom
   src/ftxui/dom/inverted.cpp
   src/ftxui/dom/linear_gradient.cpp
   src/ftxui/dom/node.cpp
+  src/ftxui/dom/italic.cpp
   src/ftxui/dom/node_decorator.cpp
   src/ftxui/dom/paragraph.cpp
   src/ftxui/dom/reflect.cpp
diff --git a/include/ftxui/dom/elements.hpp b/include/ftxui/dom/elements.hpp
--- a/include/ftxui/dom/elements.hpp
+++ b/include/ftxui/dom/elements.hpp
@@ -98,5 +98,6 @@ Element bold(Element);
 Element dim(Element);
 Element inverted(Element);
+Element italic(Element);
 Element underlined(Element);
 Element underlinedDouble(Element);
 Element blink(Element);
diff --git a/include/ftxui/screen/screen.hpp b/include/ftxui/screen/screen.hpp
--- a/include/ftxui/screen/screen.hpp
+++ b/include/ftxui/screen/screen.hpp
@@ -21,6 +21,7 @@ struct Pixel {
       : blink(false),
         bold(false),
         dim(false),
+        italic(false),
         inverted(false),
         underlined(false),
         underlined_double(false),
@@ -32,6 +33,7 @@ struct Pixel {
   bool blink : 1;
   bool bold : 1;
   bool dim : 1;
+  bool italic : 1;
   bool inverted : 1;
   bool underlined : 1;
   bool underlined_double : 1;
diff --git a/src/ftxui/dom/italic.cpp b/src/ftxui/dom/italic.cpp
new file mode 100644
--- /dev/null
+++ b/src/ftxui/dom/italic.cpp
@@ -0,0 +1,35 @@
+// Copyright 2025 Arthur Sonzogni. All rights reserved.
+// Use of this source code is governed by the MIT license that can be found in
+// the LICENSE file.
+#include <memory>
+#include <utility>
+
+#include "ftxui/dom/elements.hpp"
+#include "ftxui/dom/node.hpp"
+#include "ftxui/dom/node_decorator.hpp"
+#include "ftxui/screen/box.hpp"
+#include "ftxui/screen/screen.hpp"
+
+namespace ftxui {
+
+/// @brief Apply italics to text.
+/// @ingroup dom
+Element italic(Element child) {
+  class Impl : public NodeDecorator {
+   public:
+    using NodeDecorator::NodeDecorator;
+
+    void Render(Screen& screen) override {
+      for (int y = box_.y_min; y <= box_.y_max; ++y) {
+        for (int x = box_.x_min; x <= box_.x_max; ++x) {
+          screen.PixelAt(x, y).italic = true;
+        }
+      }
+      Node::Render(screen);
+    }
+  };
+
+  return std::make_shared<Impl>(std::move(child));
+}
+
+}  // namespace ftxui
diff --git a/src/ftxui/screen/screen.cpp b/src/ftxui/screen/screen.cpp
--- a/src/ftxui/screen/screen.cpp
+++ b/src/ftxui/screen/screen.cpp
@@ -112,1 +112,7 @@ void UpdatePixelStyle(const Screen* screen,
+  // Italics
+  if (FTXUI_UNLIKELY(next.italic != prev.italic)) {
+    ss << (next.italic ? "\x1B[3m"     // ITALIC_SET
+                       : "\x1B[23m");  // ITALIC_RESET
+  }
+
   // StrikeThrough
